### PR TITLE
Add proposals to webbjs

### PR DESCRIPTION
**Pull request by drewstone**
# Overview
Proposal type encoding/decoding for proposals from the DKG repo that were going stale. We should have them in the JS API anyway so we can keep them up-to-date for any repos that rely on them either in testing or in production.

This is meant to be paired with removing these files from the DKG repo and any old stale tests there. The PR with the removed files is: https://github.com/webb-tools/dkg-substrate/pull/347. We will need this merged/working/tested in order to get the other PR ready for final updates.

CC @salman01zp can you check to see if this will impact the relayer tests in any way as well? Maybe this helps on that front too.

### CODE REVIEW

**Comment by drewstone** [comment]
CC @nepoche I'm confused why the build keeps failing, must be missing some `.js`?

File: packages/sdk-core/src/proposals/ResourceId.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    case 0x0302:
+      return ChainIdType.KUSAMA_RELAYCHAIN;
+    case 0x0400:
+      return ChainIdType.COSMOS;
+    case 0x0500:
+      return ChainIdType.SOLANA;
+    default:
+      return ChainIdType.UNKNOWN;
+  }
+}
```
Comment (nepoche) [inline packages/sdk-core/src/proposals/ResourceId.ts:34]: Move to `typed-chain-id.ts`

File: packages/sdk-core/src/proposals/index.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+export const LE = true;
+export const BE = false;
```
Comment (nepoche) [inline packages/sdk-core/src/proposals/index.ts:18]: Is this meant to indicate the proposals should always be encoded as little endian?

**Review by nepoche — CHANGES_REQUESTED** [review]

File: packages/sdk-core/src/proposals/index.ts (hunk 2 above)
Comment (drewstone) [reply packages/sdk-core/src/proposals/index.ts:18]: No, this is a boolean flag used in certain functions to indicate the type of encoding you want. It exists on the `DataView` functions. I will add a comment.

**Review by nepoche — APPROVED** [review]

**Pull request merged**